histomicstk/cli/NucleiDetection/NucleiDetection.py

```diff
@@ -76,14 +121,141 @@ def detect_tile_nuclei(slide_path, tile_position, args, it_kwargs,
+        )
+
+    else:
+
+        tile_fgnd_frac_list = np.full(num_tiles, 1.0)
+
+    num_fgnd_tiles = np.count_nonzero(
+        tile_fgnd_frac_list >= args.min_fgnd_frac)
+
+    percent_fgnd_tiles = 100.0 * num_fgnd_tiles / num_tiles
+
+    fgnd_frac_comp_time = time.time() - start_time
+
+    print('Number of foreground tiles = {:d} ({:2f}%%)'.format(
+        num_fgnd_tiles, percent_fgnd_tiles))
+
+    print('Tile foreground fraction computation time = {}'.format(
+        cli_utils.disp_time_hms(fgnd_frac_comp_time)))
+
+    return tile_fgnd_frac_list
+
+
+def compute_reinhard_norm(args, invert_image=False, default_img_inversion=False):
+    print('\n>> Computing reinhard color normalization stats ...\n')
+
+    start_time = time.time()
+    src_mu_lab, src_sigma_lab = htk_cnorm.reinhard_stats(
+        args.inputImageFile, 0.01, magnification=args.analysis_mag,
+        invert_image=invert_image, style=args.style, frame=args.frame,
+        default_img_inversion=default_img_inversion)
+
+    rstats_time = time.time() - start_time
+
+    print('Reinhard stats computation time = {}'.format(
+        cli_utils.disp_time_hms(rstats_time)))
+    return src_mu_lab, src_sigma_lab
+
+
+def detect_nuclei_with_dask(ts, tile_fgnd_frac_list, it_kwargs, args,
+                            invert_image=False, is_wsi=False, src_mu_lab=None,
+                            src_sigma_lab=None, default_img_inversion=False):
+
     import dask
 
+    print('\n>> Detecting nuclei ...\n')
+
+    start_time = time.time()
+
+    tile_nuclei_list = []
+
+    for tile in ts.tileIterator(**it_kwargs):
+
+        tile_position = tile['tile_position']['position']
+
+        if is_wsi and tile_fgnd_frac_list[tile_position] <= args.min_fgnd_frac:
+            continue
+
+        # detect nuclei
+        cur_nuclei_list = dask.delayed(detect_tile_nuclei)(
+            args.inputImageFile,
+            tile_position,
+            args, it_kwargs,
+            src_mu_lab, src_sigma_lab, invert_image=invert_image, style=args.style,
+            default_img_inversion=default_img_inversion
+        )
+
+        # append result to list
+        tile_nuclei_list.append(cur_nuclei_list)
+
+    tile_nuclei_list = dask.delayed(tile_nuclei_list).compute()
+
+    nuclei_list = [anot
+                   for anot_list in tile_nuclei_list for anot in anot_list]
+
+    nuclei_detection_time = time.time() - start_time
+
+    print(f'Number of nuclei = {len(nuclei_list)}')
+
+    print('Nuclei detection time = {}'.format(
+        cli_utils.disp_time_hms(nuclei_detection_time)))
+    return nuclei_list
+
+
+def main(args):
+
+    # Flags
+    invert_image = False
+    default_img_inversion = False
+
+    # initial arguments
+    it_kwargs = {
+        'tile_size': {'width': args.analysis_tile_size},
+        'scale': {'magnification': args.analysis_mag}
+    }
+
     total_start_time = time.time()
 
     print('\n>> CLI Parameters ...\n')
 
-    print(args)
+    print(args, type(args))
```

Comment:
args will be a Namespace; it probably isn't useful to report this in general, so this could be reverted to just `print(args)`.

Reply:
Though maybe it'd be better to `pprint.pprint(vars(args))`

Reply:
The print args were there on the primary code repo. I kind of find it useful when running the job to have a quick check if all the parameters I've given are good. As you suggested pprint makes more sense for printing the params.